### ISSUE TITLE
Outgoing Proxy Support

### ISSUE THREAD

**Issue by nRaecheR** [searx/searx#236]
I'm trying to get my local Searx instance to work behind an outgoing proxy server. I've fiddled around with all kinds of http(s)_proxy/HTTP(S)_PROXY settings in the environment, without success.

Environment is Ubuntu Server 14.10, amd64. As incoming proxy an Apache2 instance is used. 

Is there any way to configure an outgoing proxy in Searx ?

**Comment by asciimoo** [comment]
Currently there is no HTTP(S) proxy support. As a workaround you can experiment with external transparent SOCKS proxy wrappers like `tsocks`.

**Comment by dalf** [comment]
According to http://docs.python-requests.org/en/latest/user/advanced/#proxies
HTTP(S)_PROXY are supported. 
I miss something ?

**Comment by nRaecheR** [comment]
I've tried this 

```
$ export HTTP_PROXY="http://myproxy:Port"
$ export HTTPS_PROXY="http://myproxy:Port"
$ python
>>> import requests
>>> requests.get("http://example.org")
```

and it works but Searx don't use the proxy setting from global environment. Does it use python::requests at all?

**Comment by dalf** [comment]
How do you start Searx ?

Using a virtualenv and starting searx in debug mode with `python webapp.py` the HTTP(S)_PROXY variables are used.

I guess it's an issue with uswsgi / supervisord

**Comment by nRaecheR** [comment]
Yes, it's with uswsgi and Apache as web server.

**Comment by ghost** [comment]
It will be useful if I can set HTTP(S) proxy in searx's config file.

**Comment by asciimoo** [comment]
implemented: #297
